tyssue/utils/utils.py

```diff
@@ -237,27 +245,17 @@ def ar_calculation(sheet):
     Returns
     -------
     AR: pandas series of aspect ratio for all faces.
-
     """
-    face_vertices = sheet.face_polygons(sheet.coords)
-    ar = []
-    for face_number in range(len(sheet.face_df)):
-        x_value = []
-        z_value = []
-        for i in range(len(face_vertices[face_number])):
-            x_value.append(face_vertices[face_number][i][0])
-            z_value.append(face_vertices[face_number][i][2])
-        x_value = np.array(x_value)
-        z_value = np.array(z_value)
-        major = x_value.ptp()
-        minor = z_value.ptp()
-
-        if major < minor:
-            tmp = major
-            major = minor
-            minor = tmp
-        if minor == 0:
-            ar.append(0)
-        else:
-            ar.append(major / minor)
-    return pd.Series(ar)
+    from ..geometry.sheet_geometry import SheetGeometry
+
+    projected_srce_pos = pd.DataFrame()
+    for face in np.unique(sheet.edge_df['face']):
+        projected_srce_pos = SheetGeometry.face_projected_pos(sheet, face)
```

Comment:
I don't think this is what you intended. The face_projected_pos should go in the _compute_ar function to be called for each face, you can get the face index with df['face'].iloc[0].